core/src/main/scala/kafka/api/LeaderAndIsr.scala

```diff
@@ -43,6 +44,6 @@ case class LeaderAndIsr(leader: Int,
   def newEpochAndZkVersion = newLeaderAndIsr(leader, isr)
 
   override def toString: String = {
-    Json.encode(Map("leader" -> leader, "leader_epoch" -> leaderEpoch, "isr" -> isr))
+    Json.encodeToJsonString(Map("leader" -> leader, "leader_epoch" -> leaderEpoch, "isr" -> isr.asJava))
```

Comment:
The existing code for this is weird. We should not be using json in the `toString` method.

Reply:
So do you have any suggestions to implement this overridden `toString` method except returning Json string.

Reply:
Just a normal `toString` like other classes.

Reply:
Sure, how about this: 

`s"leader $leader, leader_epoch $leaderEpoch, isr $isr"`

Reply:
Actually this seems to have all info for leader and isr 

`    s"leader $leader, leader_epoch $leaderEpoch, isr $isr, zk_version $zkVersion"
`

Reply:
You should follow the same pattern as other classes. One example is `ProducerRecord`.